Sources/Crypto/Signatures/ECDSA.swift

```diff
@@ -197,11 +229,15 @@ extension P384.Signing {
 
         /// A DER-encoded representation of the signature
         public var derRepresentation: Data {
-            #if (os(macOS) || os(iOS) || os(watchOS) || os(tvOS)) && !CRYPTO_IN_SWIFTPM_FORCE_BUILD_API
-            return self.coreCryptoDERRepresentation
-            #else
-            return self.openSSLDERRepresentation
-            #endif
+            let raw = rawRepresentation
+            let half = raw.count / 2
+            let r = Array(raw.prefix(upTo: half))[...]
+            let s = Array(raw.suffix(from: half))[...]
+
+            let sig = ASN1.ECDSASignature(r: r, s: s)
+            var serializer = ASN1.Serializer()
+            try! serializer.serialize(sig)
```

Comment:
Why is the `!` okay here?

Reply:
The `!` is ok for two reasons:

1. This is a `var`, so we can't throw even if we wanted to: crashing is our only accessible error handling mode
2. This is not being computed over arbitrary data but previously-validated data, so it should not be possible for this to fail. If it did, it would be our error.

Reply:
Thanks for the explainer!